dashboard/test/integration/redirects_test.rb

```diff
@@ -52,8 +52,8 @@ class RedirectsTest < ActionDispatch::IntegrationTest
     get '/s/frozen/lang/es'
     assert_redirected_to '/s/frozen'
 
-    get '/s/course1/stage/1/puzzle/1/lang/es'
-    assert_redirected_to '/s/course1/stage/1/puzzle/1'
+    get '/s/course1/lessons/1/levels/1/lang/es'
+    assert_redirected_to '/s/course1/lessons/1/levels/1'
```

Comment:
I added a test which tests a couple of the common urls we will be redirecting. Two other cases I thought would be good to test but found harder to do were checking a level in a lockable lesson and a lesson extra. I think this will require setting up scripts and levels that have this set up. Is it worth doing that as well?

Reply:
It looks like there are only three places where stagePosition needed to be updated so I just did it.

Reply:
I think I'm gonna try setting up a test for redirect for lockable and lesson extras later today just to be safe. Any other cases you can think of?

Reply:
I think testing the lockable redirects and 2-3 of the various script level path formats (e.g. regular script level, level group subpage, lesson extras) should suffice.

Reply:
Tests have been added. Found some paths that needed redirect routes so glad I did it. Take a look

Reply:
Done!